Sources/RegexBenchmark/BenchmarkRunner.swift

```diff
@@ -37,36 +34,43 @@ public struct BenchmarkRunner {
       let time = end.elapsedTime(since: start)
       times.append(time)
     }
-    // todo: compute stdev and warn if it's too large
-    
-    // return median time
+
     times.sort()
     let median = times[samples/2]
-    self.results.add(name: benchmark.name, time: median)
-    return median
+    let mean = times.reduce(0.0, {acc, next in acc + next.seconds}) / Double(times.count)
+    let stdev = (times.reduce(0.0, {acc, next in acc + pow(next.seconds - mean, 2)}) / Double(times.count)).squareRoot()
+    return BenchmarkResult(median, stdev, samples)
   }
   
-  public mutating func run() {
+  mutating func run() {
     print("Running")
     for b in suite {
-      print("- \(b.name) \(measure(benchmark: b))")
-    }
-  }
-  
-  public func profile() {
-    print("Starting")
-    for b in suite {
-      print("- \(b.name)")
-      b.run()
-      print("- done")
+      var result = measure(benchmark: b, samples: samples)
+      if !quiet {
+        print("- \(b.name) \(result.median) (stdev: \(Time(result.stdev)))")
+      }
+      
+      if result.stdev > Stats.maxAllowedStdev {
+        print("Warning: Standard deviation > \(Time(Stats.maxAllowedStdev)) for \(b.name)")
+        print("N = \(samples), median: \(result.median), stdev: \(Time(result.stdev))")
+        print("Rerunning \(b.name)")
+        result = measure(benchmark: b, samples: result.samples*2)
+        print("N = \(result.samples), median: \(result.median), stdev: \(Time(result.stdev))")
+        if result.stdev > Stats.maxAllowedStdev {
+          fatalError("Benchmark \(b.name) is too variant")
```

Comment:
Should this print the error instead of trapping the program?

Reply:
Being too variant generally indicates that the system is running stuff in the background and making the benchmarker variant, so in those cases I thought it would probably best to just stop everything and make the user restart it